homeassistant/components/mqtt/remote.py

```diff
@@ -0,0 +1,183 @@
+)
+
+PLATFORM_SCHEMA = (
+    mqtt.MQTT_RW_PLATFORM_SCHEMA.extend(
+        {
+            vol.Optional(CONF_DEVICE): mqtt.MQTT_ENTITY_DEVICE_INFO_SCHEMA,
+            vol.Optional(CONF_ICON): cv.icon,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_PAYLOAD_OFF, default=DEFAULT_PAYLOAD_OFF): cv.string,
+            vol.Optional(CONF_PAYLOAD_ON, default=DEFAULT_PAYLOAD_ON): cv.string,
+            vol.Optional(CONF_COMMANDS, default={}): cv.schema_with_slug_keys(
+                COMMAND_SCHEMA
+            ),
+        }
+    )
+    .extend(mqtt.MQTT_AVAILABILITY_SCHEMA.schema)
+    .extend(mqtt.MQTT_JSON_ATTRS_SCHEMA.schema)
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType, config: ConfigType, async_add_entities, discovery_info=None
+):
+    """Set up MQTT remote through configuration.yaml."""
+    await _async_setup_entity(config, async_add_entities, discovery_info)
+
+
+async def _async_setup_entity(config, async_add_entities, config_entry=None):
+    """Set up the MQTT remote."""
+    async_add_entities([MqttRemote(config, config_entry)])
+
+
+# pylint: disable=too-many-ancestors
+class MqttRemote(
+    MqttAttributes, MqttAvailability, MqttEntityDeviceInfo, RemoteDevice, RestoreEntity
+):
+    """Representation of a remote that can be controlled using MQTT."""
+
+    def __init__(self, config, config_entry):
+        """Initialize the MQTT remote."""
+        self._state = False
+        self._sub_state = None
+
+        self._state_on = None
+        self._state_off = None
+        self._optimistic = None
+        self._unique_id = config.get(CONF_UNIQUE_ID)
+
+        # Load config
+        self._setup_from_config(config)
+
+        device_config = config.get(CONF_DEVICE)
+
+        MqttAttributes.__init__(self, config)
+        MqttAvailability.__init__(self, config)
+        MqttEntityDeviceInfo.__init__(self, device_config, config_entry)
+
+    async def async_added_to_hass(self):
+        """Subscribe to MQTT events."""
+        await super().async_added_to_hass()
+
+    def _setup_from_config(self, config):
+        """(Re)Setup the entity."""
+        self._config = config
+        self._optimistic = config[CONF_OPTIMISTIC]
+        self._commands = config.get(CONF_COMMANDS)
+
+    async def async_will_remove_from_hass(self):
+        """Unsubscribe when removed."""
+        self._sub_state = await subscription.async_unsubscribe_topics(
+            self.hass, self._sub_state
+        )
+        await MqttAttributes.async_will_remove_from_hass(self)
+        await MqttAvailability.async_will_remove_from_hass(self)
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._config[CONF_NAME]
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
```

Comment:
Did I miss the subscription and callback handling for the state or is it only optimistic?

Reply:
No you didn't miss it. I will add it.